py/selenium/webdriver/edge/webdriver.py

```diff
@@ -15,51 +15,57 @@
 # specific language governing permissions and limitations
 # under the License.
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
-from selenium.webdriver.edge.service import Service
+
 from selenium.webdriver.chromium.webdriver import ChromiumDriver
+from .options import Options
+from .service import Service
 
 
 DEFAULT_PORT = 0
 DEFAULT_SERVICE_LOG_PATH = None
 
 
 class WebDriver(ChromiumDriver):
+    """
+    Controls the Microsoft Edge driver and allows you to drive the browser.
+    You will need to download either the MicrosoftWebDriver (Legacy)
+    or MSEdgeDriver (Chromium) executable from
+    https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/
+    """
 
-    def __init__(self, executable_path='MicrosoftWebDriver.exe',
-                 capabilities=None, port=DEFAULT_PORT, verbose=False,
-                 service_log_path=None, log_path=DEFAULT_SERVICE_LOG_PATH,
-                 service=None, options=None, keep_alive=False, is_legacy=True,
-                 service_args=None):
+    def __init__(self, executable_path='MicrosoftWebDriver.exe', port=DEFAULT_PORT,
+                 options=None, service_args=None,
+                 capabilities=None, service_log_path=DEFAULT_SERVICE_LOG_PATH,
+                 service=None, keep_alive=False, verbose=False):
         """
         Creates a new instance of the edge driver.
         Starts the service and then creates new instance of edge driver.
 
         :Args:
          - executable_path - Deprecated: path to the executable. If the default is used it assumes the executable is in the $PATH
-         - capabilities - Dictionary object with non-browser specific capabilities only, such as "proxy" or "loggingPref".
-           Only available in Legacy mode
          - port - Deprecated: port you would like the service to run, if left as 0, a free port will be found.
-         - verbose - whether to set verbose logging in the service. Only available in Legacy Mode
+         - options - this takes an instance of EdgeOptions
+         - service_args - Deprecated: List of args to pass to the driver service
+         - capabilities - Deprecated: Dictionary object with non-browser specific
+           capabilities only, such as "proxy" or "loggingPref".
          - service_log_path - Deprecated: Where to log information from the driver.
          - keep_alive - Whether to configure EdgeRemoteConnection to use HTTP keep-alive.
-         - service_args - Deprecated: List of args to pass to the driver service
-         - is_legacy: Whether to use MicrosoftWebDriver.exe (legacy) or MSEdgeDriver.exe (chromium-based). Defaults to True.
+         - verbose - whether to set verbose logging in the service.
          """
-        if not is_legacy:
+        if executable_path != 'MicrosoftWebDriver.exe':
+            warnings.warn('executable_path has been deprecated, please pass in a Service object',
```

Comment:
There is no import for `warnings`. You can see the [travis log](https://travis-ci.com/SeleniumHQ/selenium/jobs/296255724) of the failure